apps/studio/components/interfaces/DataWarehouse/WarehouseCollectionDetail.tsx

```diff
@@ -81,23 +124,80 @@ export const WarehouseCollectionDetail = () => {
   const results = formatResults(queryData?.result)
 
   function loadMore() {
-    setPagination({ ...pagination, offset: pagination.offset + pagination.limit })
+    setFilters({ ...filters, offset: filters.offset + filters.limit })
   }
 
   const isLoading = queryLoading || collectionsLoading || isRefetching
 
+  function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault()
+    setFilters({ ...filters, search })
+    refetch()
+  }
+
+  const queryUrl = `/project/${projectRef}/logs/explorer?q=${encodeURIComponent(
+    params.sql || ''
+  )}&source=warehouse`
+
   return (
     <div className="relative flex flex-col flex-grow h-full">
       <ShimmerLine active={isLoading} />
       <LoadingOpacity active={isLoading}>
         <div className="flex flex-col w-full">
-          <div className="flex justify-between items-center pr-3">
-            <h2 className="p-3">{collection?.name}</h2>
+          <div className="flex justify-between items-center h-12 px-5">
+            <form onSubmit={handleSubmit} className="flex items-center gap-2">
+              <h2 className="text-foreground-light">{collection?.name}</h2>
+              <Tooltip>
+                <TooltipTrigger asChild>
+                  <Button
+                    htmlType="submit"
+                    icon={<RefreshCcw />}
+                    type="text"
+                    loading={isLoading}
+                    disabled={isLoading}
+                    className="px-1.5"
+                  />
+                </TooltipTrigger>
+                <TooltipContent>Refresh</TooltipContent>
+              </Tooltip>
+              <Input
+                className="w-40"
+                size="tiny"
+                placeholder="Search by event message"
+                value={search}
+                onChange={(e) => setSearch(e.target.value)}
+              />
+              <DatePickers
+                to={filters.interval?.to}
+                from={filters.interval?.from}
+                helpers={INTERVALS}
+                onChange={(e) =>
+                  setFilters({
+                    ...filters,
+                    interval: {
+                      to: e.to || '',
+                      from: e.from || '',
+                    },
+                  })
+                }
+              />
+            </form>
             <div className="flex items-center gap-2">
               <Button asChild type={'text'}>
                 <Link href={`/project/${projectRef}/settings/warehouse`}>Access tokens</Link>
               </Button>
 
+              <Tooltip>
+                <TooltipTrigger asChild>
+                  <Button asChild className="px-1.5" type="outline" icon={<Terminal />}>
+                    <Link href={queryUrl} />
+                  </Button>
+                </TooltipTrigger>
+                <TooltipContent side="bottom" className="text-xs">
+                  Open query in Warehouse Explorer
```

Comment:
can probably trim the sql when opening into the warehouse explorer?
![image](https://github.com/user-attachments/assets/ee1ee2aa-7447-4092-b9af-7f831a44cf1c)